packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2408,6 +2434,17 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
   ///    setting [isSemanticBoundary] to true.
   bool get isSemanticBoundary => false;
 
+  /// Whether this [RenderObject] makes other [RenderObject]s previously painted
+  /// in the same semantics container unreachable for accessibility purposes.
+  ///
+  /// If `true` is returned, the [SemanticsNode]s for all siblings and cousins
+  /// of this node, that were painted before this node, are dropped from the
+  /// semantics tree up until a semantic boundary is reached.
+  ///
+  /// Paint order as established by [visitChildrenForSemantics] is used to
+  /// determine if a node is previous to this one.
+  bool get isBlockingSemanticsOfPreviouslyPaintedNodes => false;
```

Comment:
this should define how it interacts when set on a node that is also a boundary.

I suspect the best behaviour is that it acts on the "outside" of the boundary, but I'm not sure that's what the code currently does.